NPC AI Fixes and adjustments

Bobtheapple opened the pull request:
Fixed Hostile AIs to play sounds properly, using an async rather than a coroutinue to reduce allocations.  Removed many uses of transform references and replaced them with register tile references.

I also rebased MobMeleeAttack on MobMeleeAction and  FaceHuggerAi on GenericHostileAI.   These scripts had a lot of overlap, doing this simplified them greatly.  MobMeleeAction was made more configurable by subdividing CheckForTargetAction into methods for validating the target, always acting on target, acting on players, acting on npcs, and acting on tiles.  This will allow programmers to more easily add mobs that act generally like another mob, but interact differently with one group of targets.  Also, in the process some AI bugs were potentially fixed.

Fixed some health stuff from the health update.

Bobtheapple committed:
NPC AI Fixes

Fixed Hostile AIs to play sounds properly.  Removed many uses of transform references and replaced them with register tile references. Some health updates.

Rebased MobMeleeAttack on MobMeleeAction, and made MobMeleeAction more configurable.  Adjusted some targeting logic.  Rebased FaceHuggerAi on GenericHostileAI

Bobtheapple committed:
Fixes

Added includes to Cat and Corgi AI, fixed some more transforms

corp-0 commented on UnityProject/Assets/Scripts/NPC/AI/Friendly/CatAI.cs:
```diff
@@ -90,7 +91,7 @@ private MouseAI AnyMiceNearby()
 
 				if (coll.GameObject != gameObject && coll.GameObject.GetComponent<MouseAI>() != null
-				                                  && !coll.GameObject.GetComponent<LivingHealthBehaviour>().IsDead)
+				                                  && !coll.GameObject.GetComponent<LivingHealthMasterBase>().IsDead)
```
are mobs using the new health classes now? I think you need to add raceData and the list of organs, I might be wrong though.

corp-0 commented on UnityProject/Assets/Scripts/NPC/AI/MobMeleeAction.cs:
```diff
@@ -9,20 +9,20 @@
+				 "a player originally.")]
 		[SerializeField]
-		private bool targetOtherPlayersWhoGetInWay = true;
+		public bool targetOtherPlayersWhoGetInWay = true;
```
if your intention was to make these accesible to the children classes, use ``protected`` instead. If you're getting this from another place, add a public property that returns the private value.

Bobtheapple committed:
ActOnLiving Changes and public->protected

Changed ActOnLiving to be two functions, one for the old health system and one for the new.  Had to change MobMeleeAttack, in the process changed the coroutine to an Async.  Changed many public variables to protected.  Reverted cats and corgis to target based on old health system.

Bobtheapple committed:
FaceHugger Fixes

Tweaked GenericHostileAI to not force MobMeleeAttack, moved code for Face Hugging to its own FaceHugAction script.

Bobtheapple committed:
Fixes

Changed FaceHugger script order, now will face hug.  Removed some debug logs and tweaked statueai and generichostileai to use mobmeleeaction vs attack/follow, in case that mattered

Bobtheapple committed:
Update MobMeleeAttack.cs

Bobtheapple committed:
Update MobMeleeAttack.cs
